### DISCUSSION TITLE
Polymorphic category relationship (Feature idea)

### DISCUSSION THREAD

**Discussion by gorjan-mishevski** [saleor/saleor#7275] [Ideas]
I was wondering if Saleor will be able to manage a product to be part of 1 or N number of categories.
This allows for greater flexibility while creating hierarchical navigation.

For example:
Skincare (Cat) > Cream (Cat) > The Product.
Brands (Cat) > A Brand (Cat) > The Same Product.
Problem (Cat) > Acne (Cat) > The Same Product.

We got The Product by accessing from a category that contains skin care creams, but we also are able to access the same product by traversing through the brands. It seems with the current way a product is able to be part only of a single category which can be limiting in various situations. Because a brand category can contain products that are part of various different categories and filtering with product attributes is not cutting it. This seems a good idea to fill this feature gap in Saleor.

**Comment by krzysztofwolski** [comment]
ATM we are not planning to change how categories work. 

Creating different tree structures could be also achieved using Menu, where you can link the collection to the menu item.

**Comment by blueyo** [comment]
As stated by @krzysztofwolski having the option to link 1 product to N collections should achieve the same job @gorjan-mishevski  your after. Magento does not have Collections concept would results into not having N categories and not exactly knowing what is the best selling categories becuase the same product could be liked N categories.
